src/app/qgsdecorationcopyright.cpp

```diff
@@ -91,77 +107,78 @@ void QgsDecorationCopyright::run()
-    QString style = "<style type=\"text/css\"> p {color: " +
-                    QStringLiteral( "rgba( %1, %2, %3, %4 )" ).arg( mColor.red() ).arg( mColor.green() ).arg( mColor.blue() ).arg( QString::number( mColor.alphaF(), 'f', 2 ) ) + "}</style>";
-    text.setHtml( style + "<p>" + displayString + "</p>" );
-    QSizeF size = text.size();
+  int deviceHeight = context.painter()->device()->height();
+  int deviceWidth = context.painter()->device()->width();
 
-    float myXOffset( 0 ), myYOffset( 0 );
+  float xOffset( 0 ), yOffset( 0 );
 
-    // Set  margin according to selected units
-    switch ( mMarginUnit )
+  // Set  margin according to selected units
+  switch ( mMarginUnit )
+  {
+    case QgsUnitTypes::RenderMillimeters:
     {
-      case QgsUnitTypes::RenderMillimeters:
-      {
-        int myPixelsInchX = context.painter()->device()->logicalDpiX();
-        int myPixelsInchY = context.painter()->device()->logicalDpiY();
-        myXOffset = myPixelsInchX * INCHES_TO_MM * mMarginHorizontal;
-        myYOffset = myPixelsInchY * INCHES_TO_MM * mMarginVertical;
-        break;
-      }
-
-      case QgsUnitTypes::RenderPixels:
-        myXOffset = mMarginHorizontal;
-        myYOffset = mMarginVertical;
-        break;
-
-      case QgsUnitTypes::RenderPercentage:
-        myXOffset = ( ( myWidth - size.width() ) / 100. ) * mMarginHorizontal;
-        myYOffset = ( ( myHeight - size.height() ) / 100. ) * mMarginVertical;
-        break;
-
-      default:  // Use default of top left
-        break;
+      int pixelsInchX = context.painter()->device()->logicalDpiX();
+      int pixelsInchY = context.painter()->device()->logicalDpiY();
+      xOffset = pixelsInchX * INCHES_TO_MM * mMarginHorizontal;
+      yOffset = pixelsInchY * INCHES_TO_MM * mMarginVertical;
+      break;
     }
-    //Determine placement of label from form combo box
-    switch ( mPlacement )
+    case QgsUnitTypes::RenderPixels:
     {
-      case BottomLeft: // Bottom Left. myXOffset is set above
-        myYOffset = myHeight - myYOffset - size.height();
-        break;
-      case TopLeft: // Top left. Already setup above
-        break;
-      case TopRight: // Top Right. myYOffset is set above
-        myXOffset = myWidth - myXOffset - size.width();
-        break;
-      case BottomRight: // Bottom Right
-        //Define bottom right hand corner start point
-        myYOffset = myHeight - myYOffset - size.height();
-        myXOffset = myWidth - myXOffset - size.width();
-        break;
-      default:
-        QgsDebugMsg( QString( "Unknown placement index of %1" ).arg( static_cast<int>( mPlacement ) ) );
+      xOffset = mMarginHorizontal;
+      yOffset = mMarginVertical;
+      break;
     }
+    case QgsUnitTypes::RenderPercentage:
+    {
+      xOffset = ( ( deviceWidth - textWidth ) / 100. ) * mMarginHorizontal;
+      yOffset = ( ( deviceHeight - textHeight ) / 100. ) * mMarginVertical;
+      break;
+    }
+    default:  // Use default of top left
```

Comment:
Can you remove the default and swap to explicit cases for the remaining values please?